minecode/visitors/generic.py

```diff
@@ -77,3 +76,86 @@ def process_request(purl_str):
 
     if error_msg:
         return error_msg
+
+
+def map_directory_listed_package(package_url):
+    """
+    Add a directory listed `package_url` to the PackageDB.
+
+    Return an error string if any errors are encountered during the process
+    """
+    from minecode.model_utils import add_package_to_scan_queue
+    from minecode.model_utils import merge_or_create_package
+
+    packages = [p for p in info(str(package_url)) or []]
+
+    if not packages:
+        error = f"Could not find package using fetchcode: {package_url}"
+        logger.error(error)
+        return error
+
+    package_data = packages[0].to_dict()
+
+    # Remove obsolete Package fields see https://github.com/nexB/fetchcode/issues/108
+    del package_data["api_url"]
+    del package_data["license_expression"]
+    del package_data["declared_license"]
+    del package_data["root_path"]
+    del package_data["contains_source_code"]
+    del package_data["purl"]
+
+    package = PackageData(**package_data)
```

Comment:
Or may be a better way would be to have a PackageData.from_dict method that does handles and encapsulate and cleanup before the creation.
 
If you have it here something like:
```
def packagedata_from_dict(package_data):
    """
    Return a PackageData built from a `package_data` mapping.
    Ignore unknown and unsupported fields.
    """
    #.... remove all unsupported fields (possibly using introspection of PackageData using attrs) 
    return PackageData(**cleaned_package_data)   
```
and later on contribute this to SCTK

Reply:
Done